x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/audit/logfile/LoggingAuditTrail.java

```diff
@@ -1355,33 +1377,40 @@ public static void registerSettings(List<Setting<?>> settings) {
          * a singleton list of the empty string ([""]).
          */
         EventFilterPolicy(String name, Predicate<String> ignorePrincipalsPredicate, Predicate<String> ignoreRealmsPredicate,
-                Predicate<String> ignoreRolesPredicate, Predicate<String> ignoreIndicesPredicate) {
+                Predicate<String> ignoreRolesPredicate, Predicate<String> ignoreIndicesPredicate,
+                Predicate<String> ignoreActionsPredicate) {
             this.name = name;
             // "null" values are "unexpected" and should not match any ignore policy
             this.ignorePrincipalsPredicate = ignorePrincipalsPredicate;
             this.ignoreRealmsPredicate = ignoreRealmsPredicate;
             this.ignoreRolesPredicate = ignoreRolesPredicate;
             this.ignoreIndicesPredicate = ignoreIndicesPredicate;
+            this.ignorePrivilegesPredicate = ignoreActionsPredicate;
         }
 
         private EventFilterPolicy changePrincipalsFilter(List<String> filtersList) {
             return new EventFilterPolicy(name, parsePredicate(filtersList), ignoreRealmsPredicate, ignoreRolesPredicate,
-                    ignoreIndicesPredicate);
+                    ignoreIndicesPredicate, ignorePrivilegesPredicate);
         }
 
         private EventFilterPolicy changeRealmsFilter(List<String> filtersList) {
             return new EventFilterPolicy(name, ignorePrincipalsPredicate, parsePredicate(filtersList), ignoreRolesPredicate,
-                    ignoreIndicesPredicate);
+                    ignoreIndicesPredicate, ignorePrivilegesPredicate);
         }
 
         private EventFilterPolicy changeRolesFilter(List<String> filtersList) {
             return new EventFilterPolicy(name, ignorePrincipalsPredicate, ignoreRealmsPredicate, parsePredicate(filtersList),
-                    ignoreIndicesPredicate);
+                    ignoreIndicesPredicate, ignorePrivilegesPredicate);
         }
 
         private EventFilterPolicy changeIndicesFilter(List<String> filtersList) {
             return new EventFilterPolicy(name, ignorePrincipalsPredicate, ignoreRealmsPredicate, ignoreRolesPredicate,
-                    parsePredicate(filtersList));
+                    parsePredicate(filtersList), ignorePrivilegesPredicate);
+        }
+
+        private EventFilterPolicy changePrivilegesFilter(List<String> filtersList) {
+            return new EventFilterPolicy(name, ignorePrincipalsPredicate, ignoreRealmsPredicate, ignoreRolesPredicate,
+                ignoreIndicesPredicate, parsePredicate(filtersList));
```

Comment:
I think some validation of the privilege names would be very welcomed!

We can use `IndexPrivilege#resolve` and `ClusterPrivilegeResolver#resolve` for every item in the list.
This raises an interesting question. There are index and cluster privileges with the same names.
For example:
```
xpack.security.audit.logfile.events.ignore_filters.test_policy.indices: ["meta-index-*"]
xpack.security.audit.logfile.events.ignore_filters.test_policy.privileges: ["monitor"]
```
ignores events for the  `monitor` actions on the "meta-index-*" namespace, but
```
xpack.security.audit.logfile.events.ignore_filters.test_policy.privileges: ["monitor"]
```
Ignores events for the monitor cluster actions and for monitor actions on all indices.

I find it a bit surprising, that because of a name clash, some, but not all, privilege names are evaluated in both a cluster and an index context.

What if 
```
xpack.security.audit.logfile.events.ignore_filters.test_policy.privileges: ["monitor"]
```
is only evaluated in a cluster context, while
```
xpack.security.audit.logfile.events.ignore_filters.test_policy.indices: ["*"]
xpack.security.audit.logfile.events.ignore_filters.test_policy.privileges: ["monitor"]
```
is the equivalent, but evaluated in an index context.
This separation would help a lot with validation.

What do you think?

Reply:
Added validation + tests